configure.ac

```diff
@@ -432,28 +432,20 @@ SQUID_STATE_ROLLBACK(LIBATOMIC)
 AC_SUBST(ATOMICLIB)
 
 SQUID_AUTO_LIB(psapi,[Windows PSAPI.dll],[LIBPSAPI])
-AS_IF([test "x$with_psapi" != "xno"],[
+SQUID_CHECK_LIB_WORKS(psapi,[
   SQUID_STATE_SAVE(squid_psapi_state)
   CPPFLAGS="$LIBPSAPI_CFLAGS $CPPFLAGS"
   LIBS="$LIBPSAPI_PATH $LIBS"
-  AC_CHECK_LIB([psapi],[K32GetProcessMemoryInfo],[LIBPSAPI_LIBS="-lpsapi"])
-  AC_CHECK_HEADER([psapi.h],,,[
+  AC_CHECK_LIB([psapi],[K32GetProcessMemoryInfo],[
+    LIBPSAPI_LIBS="-lpsapi"
+    AC_CHECK_HEADER([psapi.h],,,[
 #if HAVE_WINDOWS_H
 #include <windows.h>
 #endif
+    ])
```

Comment:
>  If the system header file header-file is compilable, execute shell commands action-if-found, otherwise execute action-if-not-found. If you just want to define a symbol if the header file is available, consider using AC_CHECK_HEADERS instead.

If I am interpreting AC_CHECK_HEADER() documentation (quoted above) correctly, this AC_CHECK_HEADER() call does not define HAVE_PSAPI_H. The same is true for the official configure.ac code that compiles into this (after the actual header presence check):

```shell
if test "x$ac_cv_header_psapi_h" = xyes
then :

fi
```

If the above analysis is correct, then Squid never defines HAVE_PSAPI_H and never actually includes psapi.h despite attempting to do so.

This comment is just FYI, in case you thought this code does something useful. I am not requesting any changes here.

Reply:
done in [7909a08](https://github.com/squid-cache/squid/pull/1686/commits/7909a08807a1958112f4d089abb10d9058cf4d00)